ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+        return 1;
+    }
+
+    if (1 != fread_function(&nRecords, 4, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    if (1 != fread_function(&offset_primera_fitxa, 2, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    pMMBDXP->year = (short)(1900 + variable_byte);
+reintenta_lectura_per_si_error_CreaCampBD_XP:
+
+    if (n_queixes_estructura_incorrecta > 0)
+    {
+        if (!MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+        {
+            offset_fals = offset_primera_fitxa;
+            if ((offset_primera_fitxa - 1) % 32)
+            {
+                for (offset_fals = (offset_primera_fitxa - 1);
+                     !((offset_fals - 1) % 32); offset_fals--)
+                    ;
+            }
+        }
+    }
+    else
+        offset_reintent = ftell_function(pf);
+
+    if (1 != fread_function(&ushort, 2, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->reserved_1), 2, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->transaction_flag), 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->encryption_flag), 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->dbf_on_a_LAN), 12, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    if (MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+    {
+        memcpy(&pMMBDXP->nRecords, &nRecords, 4);
+        memcpy(((char *)&pMMBDXP->nRecords) + 4, &pMMBDXP->dbf_on_a_LAN, 4);
+    }
+    else
+        pMMBDXP->nRecords = nRecords;
+
+    if (1 != fread_function(&(pMMBDXP->MDX_flag), 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->CharSet), 1, 1, pf) ||
+        1 != fread_function(&(pMMBDXP->reserved_2), 2, 1, pf))
+    {
+        fclose_function(pf);
+        return 1;
+    }
+
+    // Checking for a cpg file
+    if (pMMBDXP->CharSet == 0)
+    {
+        FILE_TYPE *f_cpg;
+
+        strcpy(cpg_file, pMMBDXP->szFileName);
+        strcpy(cpg_file, reset_extension(cpg_file, "cpg"));
+        f_cpg = fopen_function(cpg_file, "r");
+        if (f_cpg)
+        {
+            char *p;
+            size_t read_bytes;
+            fseek_function(f_cpg, 0L, SEEK_SET);
+            if (11 > (read_bytes = fread_function(local_message, 1, 10, f_cpg)))
+            {
+                local_message[read_bytes] = '\0';
+                p = strstr(local_message, "UTF-8");
+                if (p)
+                    pMMBDXP->CharSet = MM_JOC_CARAC_UTF8_DBF;
+                p = strstr(local_message, "UTF8");
+                if (p)
+                    pMMBDXP->CharSet = MM_JOC_CARAC_UTF8_DBF;
+                p = strstr(local_message, "ISO-8859-1");
+                if (p)
+                    pMMBDXP->CharSet = MM_JOC_CARAC_ANSI_DBASE;
+            }
+            fclose_function(f_cpg);
+        }
+    }
+    if (MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+    {
+        memcpy(&pMMBDXP->FirstRecordOffset, &offset_primera_fitxa, 2);
+        memcpy(((char *)&pMMBDXP->FirstRecordOffset) + 2, &pMMBDXP->reserved_2,
```

Comment:
it is a bit odd to copy the content of a "reserved_2" field into BytesPerRecord.  Normally reserved means that it might be for later use, but here it is used. That field should likely be renamed to something like  FirstRecordOffsetUpperBits, or something like that. Or at least there should be some comment here and in the structure definition

Reply:
Yes, a litle bit odd.
I add this comment to the .h
 MM_BYTE reserved_2 // Used in extended DBF format to recompose BytesPerRecord
        [MM_MAX_LON_RESERVAT_2_BASE_DADES_XP];  // In MiraMon code: reservat_2